src/coreComponents/physicsSolvers/fluidFlow/RegionMultiphaseStatistics.cpp

```diff
@@ -0,0 +1,906 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file RegionMultiphaseStatistics.cpp
+ */
+
+#include "RegionMultiphaseStatistics.hpp"
+#include "constitutive/ConstitutiveManager.hpp"
+#include "constitutive/fluid/multifluid/MultiFluidBase.hpp"
+#include "constitutive/relativePermeability/RelativePermeabilityBase.hpp"
+#include "constitutive/solid/CoupledSolidBase.hpp"
+#include "physicsSolvers/fluidFlow/CompositionalMultiphaseBase.hpp"
+#include "mesh/MeshFields.hpp"
+#include "physicsSolvers/fluidFlow/FlowSolverBaseFields.hpp"
+#include "physicsSolvers/fluidFlow/CompositionalMultiphaseBaseFields.hpp"
+
+namespace geos
+{
+
+struct RegionMultiphaseStatistics::Statistics
+{
+  static constexpr integer MAX_NUM_PHASE = 3;
+  static constexpr integer MAX_NUM_COMPS = 5;
+  /// Indices for properties
+  static constexpr integer CELL_COUNT = 0;
+  static constexpr integer STATIC_PORE_VOLUME = 1;
+  static constexpr integer PORE_VOLUME = 2;
```

Comment:
Here, what do you think of grouping the indices in a C Enum? 
```suggestion
struct RegionMultiphaseStatistics::Statistics
{
  static constexpr integer MAX_NUM_PHASE = 3;
  static constexpr integer MAX_NUM_COMPS = 5;

  enum PropIds : integer {
    CELL_COUNT,
    STATIC_PORE_VOLUME,
    PORE_VOLUME,
    // [...]
    END
  };
```